lib/features/family/features/reflect/bloc/grateful_cubit.dart

```diff
@@ -49,50 +54,73 @@ class GratefulCubit extends CommonCubit<GratefulUIModel, GratefulCustom> {
 
   void _emitData() {
     //TODO map above fields to uimodels
+    //_currentRecommendations
+
     emitData(
       GratefulUIModel(
         avatarBarUIModel: GratefulAvatarBarUIModel(
           avatarUIModels: _profiles
-              .mapIndexed((
-                index,
-                profile,
-              ) =>
-                  profile.toGratefulAvatarUIModel(
-                    isSelected: index == _currentProfileIndex,
-                    hasDonated: _profilesThatDonated.contains(profile),
-                  ),)
+              .mapIndexed(
+                (
+                  index,
+                  profile,
+                ) =>
+                    profile.toGratefulAvatarUIModel(
+                  isSelected: index == _currentProfileIndex,
+                  hasDonated: _profilesThatDonated.contains(profile),
+                ),
+              )
               .toList(),
         ),
         recommendationsUIModel: null,
       ),
     );
   }
 
-  void onAvatarTapped(int index) {
+  Future<void> onAvatarTapped(int index) async {
+    emitLoading();
+
     _currentProfileIndex = index;
+    _currentRecommendations = await _gratefulRecommendationsRepository
+        .getGratefulRecommendations(_getCurrentProfile());
+
     _emitData();
   }
 
   void onRecommendationTapped(int index) {
     //TODO
     final organisation = _currentRecommendations[index];
     if (isCurrentProfileChild()) {
-      emitCustom(GratefulCustom.openKidDonationFlow(
-          profile: _getCurrentProfile(), organisation: organisation,),);
+      emitCustom(
+        GratefulCustom.openKidDonationFlow(
+          profile: _getCurrentProfile(),
+          organisation: organisation,
+        ),
+      );
     } else {
-      emitCustom(GratefulCustom.openParentDonationFlow(
-          profile: _getCurrentProfile(), organisation: organisation,),);
+      emitCustom(
+        GratefulCustom.openParentDonationFlow(
+          profile: _getCurrentProfile(),
+          organisation: organisation,
+        ),
+      );
     }
   }
 
-  void onDonated(GameProfile profile) {
+  Future<void> onDonated(GameProfile profile) async {
+    emitLoading();
```

Comment:
I would only put this in the else. Otherwise you emit the loading. Navigate to next. Can press the back button and be in a perpetual loading state.